django/forms/widgets.py

```diff
@@ -689,9 +690,22 @@ def create_option(
         self, name, value, label, selected, index, subindex=None, attrs=None
     ):
         index = str(index) if subindex is None else "%s_%s" % (index, subindex)
-        option_attrs = (
-            self.build_attrs(self.attrs, attrs) if self.option_inherits_attrs else {}
-        )
+
+        if not hasattr(self, "option_attrs"):
+            # AutocompleteSelect subclasses AutocompleteMixin first, and then Select
+            # (which subclasses this class).
+            # So the ChoiceWidget's __init__ method (which is how other Select classes
+            # get their option_attrs set) is not called.
+            # But AutocompleteSelect does call create_option, so we need to allow for
+            # a missing option_attrs attribute.
+            option_attrs = {}
+        elif self.option_attrs:
+            option_attrs = self.build_attrs(self.option_attrs)
+        elif self.option_inherits_attrs:
+            option_attrs = self.build_attrs(self.attrs, attrs)
+        else:
+            option_attrs = {}
```

Comment:
There are 3 different places that `option_attrs` is defaulted to `{}`… which may indicate a different approach is needed.

Reply:
I see only one now, so i think this comment is resolved now.